src/jdk.jlink/share/classes/jdk/tools/jlink/internal/JlinkTask.java

```diff
@@ -806,13 +809,57 @@ private static class ImageHelper implements ImageProvider {
         final boolean ignoreSigning;
         final Runtime.Version version;
         final Set<Archive> archives;
+        final Platform targetPlatform;
 
         ImageHelper(Configuration cf,
                     Map<String, Path> modsPaths,
                     ByteOrder order,
                     Path packagedModulesPath,
-                    boolean ignoreSigning) throws IOException {
-            this.order = order;
+                    boolean ignoreSigning,
+                    boolean verbose,
+                    PrintWriter log) throws IOException {
+            if (order != null) {
+                this.order = order;
+                String targetPlatformVal = readJavaBaseTargetPlatform(cf);
```

Comment:
This should also do the fast-path check `isJavaBaseFromCurrentPlatform(javaBasePath)` as in line 838-860.  I suggest to refactor line 838-860 to return the target platform (either runtime platform or from java.base in the given module path).   

I also suggest to include `endianness` in the `Platform` record class.

I sent you a patch offline with these suggestions.

Reply:
Thank you Mandy for the review and the improved patch. I've applied it and tests continue to pass. I've updated the PR with these changes.